ocaml/typing/typeclass.ml

```diff
@@ -274,9 +274,11 @@ let type_constraint val_env sty sty' loc =
 let make_method loc cl_num expr =
   let open Ast_helper in
   let mkid s = mkloc s loc in
-  Exp.fun_ ~loc:expr.pexp_loc Nolabel None
```

Comment:
I almost wonder whether `Ast_helper.Exp.fun_` should get a new implementation in terms of the new syntax. Do you think that would cause trouble? It would seem to offer some future-proofing.

Reply:
Jane Syntax uses `Ast_helper.Exp.fun_`, so I think it would cause trouble. I went a different route by putting an alert on `Exp.fun_` and `Exp.function_` so it's easy to audit places we've used them and so the alert text can redirect people to Jane Syntax.